src/crystal/system/win32/event_loop_iocp.cr

```diff
@@ -34,37 +35,62 @@ class Crystal::Iocp::EventLoop < Crystal::EventLoop
     iocp
   end
 
-  # Runs the event loop.
+  # Runs the event loop and enqueues the fiber for the next upcoming event or
+  # completion.
   def run_once : Nil
+    # Pull the next upcoming event from the event queue. This determines the
+    # timeout for waiting on the completion port.
+    # OPTIMIZE: Implement @queue as a priority queue in order to avoid this
+    # explicit search for the lowest value and dequeue more efficient.
     next_event = @queue.min_by(&.wake_at)
```

Comment:
Per @HertzDevil suggestion, so we fix this for good:
```suggestion
    next_event = @queue.min_by?(&.wake_at)
```

Reply:
I'd suggest to merge that as a different change to keep this one free of semantic changes.